.pre-commit-config.yaml

```diff
@@ -0,0 +1,14 @@
+repos:
```

Comment:
I am unclear what this file is for or why it needs to be committed?

Reply:
This file is to help format the changed files prior to committing
Committing this ensures that the code changes will pass most linting checks after a push.
We use this for pymatgen:
https://github.com/materialsproject/pymatgen/blob/master/.pre-commit-config.yaml

If you do not install the hook it does nothing but if it's installed it's just a nice little creature comfort thing for developers.